### DISCUSSION TITLE
DV-Multiplayer v0.1.9.0 Beta

### DISCUSSION THREAD

**Discussion by AMacro** [AMacro/dv-multiplayer#54] [Announcements]
# Changes / Fixes

With the release of Derail Valley B99 we have been forced to release a compatibility update. It is expected that there will be many yet-to-be-discovered bugs as well as a number of new features to be implemented. This release is expected to be a bridging release and may be less stable than previous released. Please report any bugs and include both the host and client `player.log` files in your report.

## UI
- Added ping times to the Server Browser
- Added LAN discovery to the Server Browser - look for the "house" icon
- Fixed an issue with the maximum length of the server name
- Added localisation for Host Game page and Server Browser - currently English and Simplified Chinese supported, more languages to come!
- When connected to a server the Save / Load game buttons have been removed, as well as the tutorial button
- When conencted to a server, exiting the session or quitting the game will now prompt to disconnect, but not prompt to save.

![image](https://github.com/user-attachments/assets/0c79fe43-a1eb-4d58-a765-5b75dfe4f471)

## Game
- B99 compatibility fixes

# Known Bugs
- The new "tow a loosely-hitched" carriage is causing some issues, we strongly recommend only towing with the hitch tightened, otherwise a de-sync will occur.
- Clients are unable to self service locomotives - self service objects (e.g. pipes, chutes, plugs, nozzles) not synced
- Horns, whistles and steam loco reversers can play up when more than one player is in the loco; this is not a breaking issue, only a mild inconvenience.
- MU chains behave the sane as the horns and whistles, playable but mildly annoying
- Shop, world items, inventory for players other than the host not yet synced
- Many other minor bugs and inconveniences, which we'll work through as we tick off the bigger issues.
- New game features including paint themes and "Demonstrator vehicles" are not completely sync'd and will be worked on over the coming releases.

# Features in progress
- Job sync - This is on hold, pending release of item sync
- Item sync - good progress has been made; there were a few outstanding issues we were working on prior to B99's release. Major updates in B99 (many new "smart" items and a new mounting system) will delay the release of this feature
- UPnP
- NAT punching

<hr /><em>This discussion was created from the release <a href='https://github.com/AMacro/dv-multiplayer/releases/tag/v0.1.9.0-Beta'>DV-Multiplayer v0.1.9.0 Beta</a>.</em>

**Comment by SeanJAnimations** [comment]
Hello! First just wanted to say, thanks SO much for getting this out SO quickly! Just wanted to let ya know about a few things. When clients put the chain over the hook for coupling. The air is automatically tied in on the host end, and, something else. Clients can actually take jobs, will see the paper, and can turn them in. However the machine does not dispense any money the client can see. Hosts can see the money, but clients can not, so I'm not sure what's goin' on there. Not sure if clients can load/unload cars for shunting jobs they take. But yeah, figured I'd let ya know of those findings. Thank y'all SO much for your efforts. It is truly appreciated!! :)

**Reply by AMacro** [reply]
Hey @SeanJAnimations,

That's some great feedback.
This weekend we'll be tackling a bunch of bugs, we probably won't squash them all but hoping to get a couple sorted and, fingers crossed, get a release out.

The mathematical model for the brakes has changed a fair bit, as has the logic for how the couplings work, so still tackling that one - we been doing some investigation on that one and a have an idea on how to tackle, but not had the opportunity to implement and test just yet. 

I think the money issue may be related to not having the item sync completed yet, so might be a little while before it gets fixed.

Big updates like B99 tend to throw progress back a bit, so we're in catchup mode - not complaining though, the update is truly awesome!

Please keep up the reports and definitely post logs if there's weird behaviour (if for any reason you want to keep a log private, you can DM me on Discord)

**Comment by Atheros524** [comment]
This happens when trying to connec to a friends session

![Screenshot 2025-01-14 195545](https://github.com/user-attachments/assets/45c6c33d-0b7c-433d-b049-80eeef2eda9d)
